amdahl/amdahl.py

```diff
@@ -124,6 +160,7 @@ def parse_command_line():
 def amdahl():
     """Amdahl's law illustrator (with fake work)"""
     rank = MPI.COMM_WORLD.Get_rank()
+    random.seed(int(time.time()) + rank)
```

Comment:
```suggestion
    # Ensure that all ranks use an unique seed when generating random numbers
    random.seed(int(time.time()) + rank)
```
For an HPC novice this an easily overlooked but very important step, worth calling it out